src/sources/http_server.rs

```diff
@@ -133,6 +133,12 @@ pub struct SimpleHttpConfig {
     #[configurable(metadata(docs::examples = "vector_http_path"))]
     path_key: OptionalValuePath,
 
+
+    /// The event key in which the remote ip is stored.
```

Comment:
Note that you'll need to regenerate the CUE doc files for the website documentation rendering.

Run:

```
$ make generate-component-docs
```

, committing the changes.

Reply:
Thanks! will do that.

Reply:
Hey, I noticed something. the command description for `generate-component-docs` is:
```
generate-component-docs: ## Generate per-component Cue docs from the configuration schema.
	${MAYBE_ENVIRONMENT_EXEC} cargo build $(if $(findstring true,$(CI)),--quiet,)
	target/debug/vector generate-schema > /tmp/vector-config-schema.json 2>/dev/null
	${MAYBE_ENVIRONMENT_EXEC} cargo vdev build component-docs /tmp/vector-config-schema.json \
		$(if $(findstring true,$(CI)),>/dev/null,)
```

what if I am using a shared target directory? I faced this issue, as my build artifacts were stored in a different directory than `target/debug`. I don't think anyone will face this issue, but just wanted to let you know.

Reply:
I am also facing issue while running the `cue` command.

```
in `<main>': undefined method `filter_map' for #<Hash:0x0000000133827d58> (NoMethodError)
Did you mean?  filter
```

Reply:
> Author

This is resolved, I unset my common target, but facing issues with `cue`. Can you tell me which stable version of cue you folks are using?

Reply:
> Note that you'll need to regenerate the CUE doc files for the website documentation rendering.
> 
> Run:
> 
> ```
> $ make generate-component-docs
> ```
> 
> , committing the changes.

This is resolved, I unset my common target, but facing issues with cue. Can you tell me which stable version of cue you folks are using?

Reply:
> what if I am using a shared target directory? I faced this issue, as my build artifacts were stored in a different directory than target/debug. I don't think anyone will face this issue, but just wanted to let you know

Ah good observation. I don't recall this coming up before. I will open a tracking issue.

https://github.com/vectordotdev/vector/issues/19843

Reply:
> Can you tell me which stable version of cue you folks are using?

Ah. I will regenerate the docs for you here. If you don't already have CUE installed. I will spare you this misery.

If you continue to encounter the need for this in future PRs, I believe we are using `v0.4.2`.

Reply:
Actually, I ran into some weirdness trying to push to your branch. If you don't mind you can try installing CUE.

If you run into issues, I can come back to this 🙏

Reply:
:crying_cat_face: As all issues go, its a simple issue of it works on my machine, didn't work on my macbook as ruby version 2.* doesn't support the functionality of `filter_map`, tried on my Linux box, worked like a charm. 

And some how, macos isn't easy enough to allow me to update the version of the ruby that is currently installed.

Any who, I have done the required changes, please let me know, if anything is left.

Reply:
😢  I'm sorry you ran into this. We also have similar struggles dealing with developing on OSX and the CI runners using linux. Thank you for pushing past that 🙏